python/ray/data/read_api.py

```diff
@@ -473,6 +478,8 @@ def read_json(
             be able to resolve file metadata more quickly and/or accurately.
         partition_filter: Path-based partition filter, if any. Can be used
             with a custom callback to read only selected partitions of a dataset.
+            By default, this filters out any files paths that do not contain the
+            .json file extension.
```

Comment:
What about json.gz?

Reply:
It's a containment check, not an exact matcheck: https://github.com/ray-project/ray/blob/40c196c361e3727ab6c3d90155849bceec72bc8f/python/ray/data/datasource/file_based_datasource.py#L158

This means that `.json.gz` would match, as would `.gz.json`.

This is also why it says "paths that do not containg the .json file extension" rather than "paths that do not end with the .json file extension"

Reply:
I see, this is ambiguous from my reading. Can you change the phrasing to `filters out file paths that do not end with .json.*`?

Reply:
That's also not accurate, since we want to supported reading `.gz.json` files as well (I've seen the compression extension come before the file extension in the wild), but `.json*` would not match `.gz.json`.

Reply:
Try 2: `filters out files with extensions not matching .*json*`?

Reply:
Also not quite right, how about `filters out files with extensions not matching *.json*`?

Reply:
SGTM